src/test/kotlin/vonage/spring/VerifiedSmsHandlerTest.kt

```diff
@@ -0,0 +1,55 @@
+package vonage.spring
+import com.vonage.client.auth.RequestSigning
+import com.vonage.client.auth.hashutils.HashUtil
+import org.apache.http.NameValuePair
+import org.apache.http.message.BasicNameValuePair
+import org.junit.Test
+import org.springframework.mock.web.MockHttpServletRequest
+import org.springframework.mock.web.MockHttpServletResponse
+import java.time.Instant
+import java.util.HashMap
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+class VerifiedSmsHandlerTest {
+
+    private fun constructDummyRequest() : HttpServletRequest
+    {
+        val request = MockHttpServletRequest()
+        val params = HashMap<String, Array<String>>()
+        val nameValuePair = ArrayList<NameValuePair>(3)
+        params.put("a", Array(1){"alphabet"})
+        params.put("b", Array(1){"bananas"})
+        params.put("timestamp", Array(1){ Instant.now().epochSecond.toString()})
+        nameValuePair.add(BasicNameValuePair("a","alphabet"))
```

Comment:
Looks like this `nameValuePair` can be removed and params can be added directly to the request - similar as in other tests

Reply:
`constructSignatureForRequestParameters` mutates the list of nameValuePairs in this (hence why it's declared separately)